### ISSUE TITLE
[NSFW] [New Website request] Add support nhentai and hanime 

### ISSUE THREAD

**Issue by prorounak2005** [ytdl-org/youtube-dl#29962]
- [ ] I'm reporting a new site support request
- [ ] I've verified that I'm running youtube-dl version **2021.06.06**
- [ ] I've checked that all provided URLs are alive and playable in a browser
- [ ] I've checked that none of provided URLs violate any copyrights
- [ ] I've searched the bugtracker for similar site support requests including closed ones

## Example URLs

Nhentai website link - https://nhentai.net/
Hanime website link - https://hanime.tv/

- Hanime Single video: https://hanime.tv/videos/hentai/jashin-shoukan-inran-kyonyuu-oyako-ikenie-gishiki-2
- Nhentai : https://nhentai.net/g/372840/

## Description

Nhentai is a website where we can read doujinshis and hanime is the website where we can watch hentai.

**Comment by dirkf** [comment]
In the Hanime example, the JSON being assigned to JS variable `window.__NUXT__` needs to be captured. Then metadata is in the `.state.data.video.hentai_video` object of the JSON, and streaming links are in `.state.data.video.videos_manifest`.

**Comment by tryoutuser** [comment]
why would you need yt-dl for nhentai. it doesn't even host videos.
ever heard of HDoujin downloader? recommended to use that instead.
if you're on phone, use Hentoid (can be found on github) to download doujinshis.

**Comment by panmarco83** [comment]
This should be pretty straight forward, when given a url to a video, the url to a m3u8 "playlist" can be found in the JS variables.
if I run the the meu8 url with YT, it downloads the video from the multiple parts/slices in the m3u8 and outputs it to a video file with a large file name based on the hash/donno of the m3u8 file path.

**Comment by prorounak2005** [comment]
> In the Hanime example, the JSON being assigned to JS variable `window.__NUXT__` needs to be captured (this is used by the NuxtJS framework to pass the page state from the server to the browser). Then metadata is in the `.state.data.video.hentai_video` object of the JSON, and streaming links are in `.state.data.video.videos_manifest`.

I cannot understand anything. I am not a computer nerd and I am noob. :\

**Comment by prorounak2005** [comment]
![Screenshot from 2021-09-17 01-19-51](https://user-images.githubusercontent.com/69617708/133676683-ff29a9f1-f02b-40f3-9bd8-6eca0082d1ea.png)

I am getting this error

**Comment by prorounak2005** [comment]
The url is not supported :(

**Comment by prorounak2005** [comment]
> why would you need yt-dl for nhentai. it doesn't even host videos.
> ever heard of HDoujin downloader? recommended to use that instead.
> if you're on phone, use Hentoid (can be found on github) to download doujinshis.

Yup no need for nhentai. I can use hakuneko and aniyomi but I need hanime.

**Comment by dirkf** [comment]
Better to paste the text output inside \`\`\`code tags\`\`\`.

Unfortunately supporting your anime site in yt-dl needs someone to write Python code, as described in the Developer Instructions in the fine manual, so you either have to learn, or hope that someone else is interested.

**Comment by panmarco83** [comment]
Well, I could probably do it, but I'd need specs on what to use.
I've built things like this before so I know how, but I'm guessing there is a procedure you would prefer used ?

**Comment by dirkf** [comment]
In the fine manual, see the section [Adding support for a new site](https://github.com/ytdl-org/youtube-dl#user-content-adding-support-for-a-new-site). But also consider whether the site meets the criteria listed in [Can you add support for this anime video site, or site which shows current movies for free?](https://github.com/ytdl-org/youtube-dl#user-content-can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free).

**Comment by panmarco83** [comment]
Okay, I can't answer that, if someone can find out if it is legal, then I can have at it.

**Comment by prorounak2005** [comment]
U can alsways download hentai from hanime for free. I read all the rules and faq :+1:
